File: app/src/main/java/com/prolocity/patchtracker/data/TeamMember.kt
package com.prolocity.patchtracker.data

import androidx.room.Entity
import androidx.room.ForeignKey
import androidx.room.Index

@Entity(
    tableName = "team_members",
    primaryKeys = ["teamId", "playerId"],
    foreignKeys = [
        ForeignKey(entity = Team::class, parentColumns = ["id"], childColumns = ["teamId"], onDelete = ForeignKey.CASCADE),
        ForeignKey(entity = Player::class, parentColumns = ["id"], childColumns = ["playerId"], onDelete = ForeignKey.CASCADE)
    ],
    indices = [Index("playerId")]
)
data class TeamMember(
    val teamId: Long,
    val playerId: Long,
    // Slot index (0-7) the player was assigned to. Position 0 is the team captain.
    val position: Int
)

File: ios/PatchTracker/Data/Models/TeamMember.swift
import Foundation
import SwiftData

/// Join row placing a `Player` in a `Team`'s roster slot. `position` (0–7) is the slot index
/// assigned in the team edit screen; slot 0 is the captain. Deleting either the team or the
/// player cascades this row away (Android `team_members` FKs, both `onDelete = CASCADE`).
@Model
final class TeamMember {
    var position: Int
    var team: Team?
    var player: Player?

    init(position: Int, team: Team?, player: Player?) {
        self.position = position
        self.team = team
        self.player = player
    }
}
